README.md

```diff
@@ -4,13 +4,248 @@ NATS Streaming is an extremely performant, lightweight reliable streaming platfo
+
+The number of sub-directories, which again correspond to channels, can be limited by the configuration parameter `-max_channels`. When the limit is reached, any new subscription or message published on a new channel will produce an error.
+
+On a given channel, the number of subscriptions can also be limited with the configuration parameter `-max_subs`. A client that tries to create a subscription on a given channel (subject) for which the limit is reached will receive an error.
+
+Finally, the number of stored messages for a given channel can also be limited with the parameter `-max_msgs` and/or `-max_bytes`. However, for messages, the client does not get an error when the limit is reached. The oldest messages are discarded to make room for the new messages.
+
+### Store Interface
+
+Every store implementation follows the [Store interface](https://github.com/nats-io/nats-streaming-server/blob/master/stores/store.go).
+
+The main interace is the `Store` which the server will create a unique instance of. From a store, the server can add/delete clients, create/lookup channels, etc...
+
+Creating/looking up a channel will return a `ChannelStore`, which points to two other interfaces, the `SubStore` and `MsgStore`. These stores handle, for a given channel, subscriptions and messages respectiverly.
+
+If you wish to contribute to a new store type, your implementation must implement all these interfaces. For stores that allow recovery (such as file store as opposed to memory store), there are additional structures that have been defined and that a store constructor should return. This allows the server to reconstruct its state on startup.
+
+The memory and the provided file store implementations both use a common store implementation to avoid code duplication. You can certaintly take advantage of that.
+
+## Building
+
+This code currently requires at least version 1.5 of Go, but we encourage the use of the latest stable release. Information on installation, including pre-built binaries, is available at http://golang.org/doc/install. Stable branches of operating system packagers provided by your OS vendor may not be sufficient.
+
+Run `go version` to see the version of Go which you have installed.
+
+Run `go build` inside the directory to build.
+
+Run `go test ./...` to run the unit regression tests.
+
+A successful build run produces no messages and creates an executable called `nats-streaming-server` in this directory. You can invoke that binary, with no options and no configuration file, to start a server with acceptable standalone defaults (no authentication, memory store).
+
+Run go help for more guidance, and visit http://golang.org/ for tutorials, presentations, references and more.
+
+## Clients
+
+Two clients are initially supported by Apcera. We will add in the future and encourage community contribution.
```

Comment:
Currently there are two sample NATS Streaming clients, both supported by Apcera. We will be adding additional supported streaming clients in the future, and encourage community-contributed clients.